Add an option for displaying all issues

ianyh opened the pull request:
Fix for #148 

General approach was to make the issues view controller polymorphic, and then use the table footer view to deal with the transition. Screenshots below.

![simulator screen shot may 11 2017 8 07 52 pm](https://cloud.githubusercontent.com/assets/212270/25977037/9bdcf136-3686-11e7-8465-c23548d1f7de.png)
![simulator screen shot may 11 2017 8 07 56 pm](https://cloud.githubusercontent.com/assets/212270/25977036/9bdc05fa-3686-11e7-8671-b241745eae3d.png)

mccarron commented:
Great implementation. Clean and re-uses a lot of what we have already. 

Could you make the table footer color match the rest of the rows? It seems too harsh in that flat white. Maybe a slightly larger font too?

subdigital commented on FiveCalls/FiveCalls/FetchIssuesOperation.swift:
```diff
@@ -11,15 +11,17 @@ import Foundation
     var issuesList: IssuesList?
 
-    init(location: UserLocation?) {
+    init(location: UserLocation?, shouldFetchAllIssues: Bool) {
```
Not sure I'm a fan of a boolean for this. Something a little less prescriptive (from the caller's perspective) might look like this:

```swift
enum IssueQuery {
    case all
    case top
}
```

Then the constructor could accept this type instead, so we'd call it like this:

```swift
FetchIssuesOperation(userLocation, query: .top)

// or

FetchIssuesOperation(userLocation, query: .all)
```

Going even further, perhaps we could move the URL creating into the enum, so the `.all` case would append the `inactive=true` query param.

Not sure if the last part fits, but just spitballing.

Doing this would give us flexibility to add more query variations in the future as well without modifying the constructor parameters.

subdigital commented on FiveCalls/FiveCalls/IssuesViewController.swift:
```diff
@@ -72,12 +79,15 @@ class IssuesViewController : UITableViewController {
     
     override var preferredStatusBarStyle: UIStatusBarStyle {
-        return .lightContent
+        return shouldFetchAllIssues ? .default : .lightContent
```
We can remove this entire method. The initial VC is actually the `IssuesContainerViewController`, which contains the blurred header thing. This VC doesn't override that method, so it's `.default` anyway.

When we push the more issues instance, the navigation controller will ask the top VC for the style, but it would be `.default` anyway.

subdigital commented on FiveCalls/FiveCalls/IssuesViewController.swift:
```diff
@@ -16,6 +16,13 @@ protocol IssuesViewControllerDelegate : class {
+
+    @IBInspectable var shouldFetchAllIssues: Bool = false {
+        didSet {
+            issuesManager.shouldFetchAllIssues = true
```
I'm not a fan of setting a property and then later firing a method, and the method's behavior is dependent on this (especially since these two steps are separated physically in the code).

I'd like to move the logic into the `fetchIssues` method directly.  We can leverage the same enum for this.

subdigital commented on FiveCalls/FiveCalls/IssuesViewController.swift:
```diff
@@ -72,12 +79,15 @@ class IssuesViewController : UITableViewController {
         logs = ContactLogs.load()
+        if shouldFetchAllIssues {
+            navigationController?.setNavigationBarHidden(false, animated: animated)
+        }
```
I'm tempted to suggest a subclass for this.  The `AllIssuesViewController` could have all of these methods defined as simple overrides, including the fetch call.

@mccarron thoughts on this?

ianyh replied:
I considered this and opted against it. It felt unnecessarily heavy-weight. I am generally pretty averse to inheritance, though.

subdigital commented:
@ianyh Thanks so much for putting this together. I've been really busy with other things and haven't been able to carve out much time for 5 Calls this week. Let me know if you want to pair on some of the above changes, I'd be game for it.

ianyh commented:
Changed the issues manager to take a query as part of the fetch call, which encapsulates the polymorphic behavior. It does feel like a cleaner interface now.

I also changed the button to be a cell, which is styled closer to the other cells. I tried styling the button itself and still using the footer view, but it didn't look right without the separator. Screenshot below.

![simulator screen shot may 12 2017 11 06 09 pm](https://cloud.githubusercontent.com/assets/212270/26022028/a031b2b2-3767-11e7-8d00-0f88678bc8df.png)
